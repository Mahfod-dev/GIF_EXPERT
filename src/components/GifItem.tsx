import { FC } from 'react';
import PropTypes from 'prop-types';

type ImageProps = {
	title: string;
	url: string;
};

const GifItem: FC<ImageProps> = ({ title, url }: ImageProps) => {
	return (
		<div className='card'>
			<img src={url} alt={title} />
			<p>{title}</p>
		</div>
	);
};

GifItem.propTypes = {
	title: PropTypes.string.isRequired,
	url: PropTypes.string.isRequired,
};

export default GifItem;
