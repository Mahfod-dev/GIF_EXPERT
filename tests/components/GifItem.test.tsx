import React from 'react';
import { render, screen } from '@testing-library/react';
import GifItem from '../../src/components/GifItem';

describe('render <GifItem/>', () => {
	const title = 'Saitama';
	const url = 'https://one-punch.com/saitama.jpg';

	test('should render snapshot', () => {
		const { container } = render(<GifItem title={title} url={url} />);

		expect(container).toMatchSnapshot();
	});

	test('should render url into img', () => {
		render(<GifItem title={title} url={url} />);

		const { src, alt } = screen.getByRole('img') as HTMLImageElement;

		expect(src).toBe(url);
		expect(alt).toBe(title);
	});

	test('should render title into GifItem', () => {
		render(<GifItem title={title} url={url} />);

		expect(screen.getByText(title)).toBeTruthy();
	});
});
