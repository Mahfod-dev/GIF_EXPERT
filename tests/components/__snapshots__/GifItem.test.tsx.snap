// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <GifItem/> should render snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
